### PR TITLE
Add EvaluatedItemCheckData

### PR DESCRIPTION
**Pull request by JanKrivanek**
### Context
Adding EvaluatedItemCheckData to BuildCheck OM - to prepare for some sample Item data/metadata Checks implementation

### Testing
The leveraging checks:
 * https://github.com/dotnet/msbuild/pull/10955
 * (TBD - EmbeddedResource should have Culture)

### CODE REVIEW

**Comment by maridematte** [comment]
Do we have plans to remove `ParsedItemsCheckData` in the future? Or are we just going to keep it as an obsolete class?

**Comment by JanKrivanek** [comment]
> Do we have plans to remove `ParsedItemsCheckData` in the future? Or are we just going to keep it as an obsolete class?

I'd give possible users at least 2 previews to use the new OM. So probably Feb/Mar next year - for 10.0, possibly for 9.0.3xx as well

**Comment by maridematte** [comment]
I think adding some documentation or comment somewhere would interesting, so in the future we remember to remove the obsolete methods. Otherwise it might end-up like our Deprecated folder.

File: src/Build/BuildCheck/API/IBuildCheckRegistrationContext.cs [hunk 1]
```diff
@@ -9,8 +9,11 @@ public interface IBuildCheckRegistrationContext
 {
     void RegisterEvaluatedPropertiesAction(Action<BuildCheckDataContext<EvaluatedPropertiesCheckData>> evaluatedPropertiesAction);
 
+    [Obsolete("Use RegisterEvaluatedItemsAction to obtain evaluated items of a project.", false)]
```
Comment (maridematte) [inline src/Build/BuildCheck/API/IBuildCheckRegistrationContext.cs:12]: I think we should add to the message when we plan to deprecate it? Both for users to be aware and as a reminder for us to clean up the code.

**Review by maridematte — APPROVED** [review]

**Review by YuliiaKovalova — APPROVED** [review]

**Review by SimaTian — APPROVED** [review]

File: src/Build/BuildCheck/API/IBuildCheckRegistrationContext.cs (hunk 1 above)
Comment (JanKrivanek) [reply src/Build/BuildCheck/API/IBuildCheckRegistrationContext.cs:12]: Let's stay with a tracking bug - we cannot (and actually should not) publicly promisse any timelines

**Pull request merged**